src/packages/auth/functions.ts

```diff
@@ -151,28 +151,36 @@ export async function login(username: string, password: string): Promise<AuthRes
         token: null,
       };
     }
-  } else {
+  } else if (AUTH_TYPE === 'none') {
     const { allowed_roles, default_role } = await getUserRoles(username, DEFAULT_ROLE_NO_AUTH, ALLOWED_ROLES_NO_AUTH);
     return {
       message: 'Authentication is disabled',
       success: true,
       token: generateJwt(username, default_role, allowed_roles),
     };
+  } else {
+    const message = 'user + pass login is not supported by current Gateway AUTH_TYPE';
+    logger.error(message);
+    return {
+      message,
+      success: false,
+      token: '',
+    }
   }
 }
 
 export async function session(authorizationHeader: string | undefined): Promise<SessionResponse> {
   const { AUTH_TYPE } = getEnv();
 
-  if (AUTH_TYPE === 'cam') {
+  if (AUTH_TYPE === 'none') {
+    return { message: `Authentication is disabled`, success: true };
+  } else {
     const { jwtErrorMessage, jwtPayload } = decodeJwt(authorizationHeader);
```

Comment:
I know the code currently doesn't check the token in the `AUTH_TYPE = none` case, but looking at `decodeJWT`, I think that behavior is incorrect. Even with Auth disabled, the Gateway should still be verifying that the token isn't expired/improperly signed, as either being the case will cause issues when it's used to communicate with Hasura.

Reply:
Good point, I suppose the `AUTH_TYPE` switch can just be removed from this function entirely.

Reply:
change has been made!